checks/apps/sphexa/sphexa_uenv.py

```diff
@@ -0,0 +1,155 @@
+# Copyright Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+import os
+import reframe as rfm
+import reframe.utility.sanity as sn
+import reframe.utility.udeps as udeps
+
+
+@rfm.simple_test
+class sphexa_build(rfm.RunOnlyRegressionTest):
+    descr = 'Clone and Build SPHEXA'
+    maintainers = ['SSA']
+    valid_systems = ['+uenv']
+    valid_prog_environs = ['+cuda']
```

Comment:
```suggestion
    valid_systems = ['+remote']
    valid_prog_environs = ['+cuda +uenv']
```
Do we want to run it on everything that simply has cuda? Or on uenvs that have cuda?

Reply:
https://github.com/eth-cscs/cscs-reframe-tests/pull/292/commits/d6a35fd59787e12655b5e9cfb61ab33ba4b11ee7

Reply:
we could create a uenv for sphexa but for now yes.

Reply:
on santis, do you get the same ?

```
uenv start prgenv-gnu/24.11:v1 -v default
ninja --version
# 1.12.1
```

Reply:
> On daint with normal+PrgEnv-gnu, normal+PrgEnv-cray (without uenv) 

well since the check is called sphexa_uenv.py, I would not run without uenv.